### PR TITLE
Init project structure

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR initiates the project with "buildSrc", "client", "model", and "server" modules.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
```
Comment (armiol) [inline gradle/wrapper/gradle-wrapper.properties:3]: Let's use the Gradle version which is currently set for, say, `kanban`.

There are some changes in Gradle 8, which are not yet compatible with Spine 1.x.

File: client/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+plugins {
+    java
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(15))
```
Comment (armiol) [inline client/build.gradle.kts:8]: Spine 1.x supports Java 8 due to some issues related to compilation. Let's use that.

File: model/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+plugins {
+    java
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(15))
```
Comment (armiol) [inline model/build.gradle.kts:8]: We don't want the copy-paste like this one in each module.

Have you seen how it's done in `kanban`? Please check out [its build script](https://github.com/spine-examples/kanban/blob/master/build.gradle.kts) (and the script-plugins it references, they are in its `buildSrc/kotlin` folder).

File: settings.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/*
```
Comment (armiol) [inline settings.gradle.kts:1]: We don't want the default "init" comment section here. It tells nothing.

Use the Apache 2.0 license headers instead. Again, please see how it's done in other examples, such as `kanban`.

File: settings.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+ * This file was generated by the Gradle 'init' task.
+ *
+ * The settings file is used to specify which projects to include in your build.
+ *
+ * Detailed information about configuring a multi-project build in Gradle can be found
+ * in the user manual at https://docs.gradle.org/8.0/userguide/multi_project_builds.html
+ */
+
+rootProject.name = "shareaware"
```
Comment (armiol) [inline settings.gradle.kts:10]: I would think, we have it camel-cased: `ShareAware`.

File: .idea/codeStyles/Project.xml [hunk 6]
```diff
@@ -0,0 +1,99 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (armiol) [inline .idea/codeStyles/Project.xml:1]: I am not able to verify it quickly, but all of our projects use pretty much the same standardized codestyle configuration. Again, please see `kanban` for reference.

**Review by armiol — CHANGES_REQUESTED** [review]
@Artem-Semenov-dev please see my comments.

So far, what I can see is that we don't re-use any ideas from the rest of our examples. But we should.

@Vladyslav-Kuksiuk FYI.

File: .idea/codeStyles/Project.xml (hunk 6 above)
Comment (Artem-Semenov-dev) [reply .idea/codeStyles/Project.xml:1]: I have copied it from kanban, not sure what can be wrong.

**Review by armiol — APPROVED** [review]
@Artem-Semenov-dev LGTM.

**Pull request merged**